core/controllers/acl_decorators_test.py

```diff
@@ -1632,6 +1634,150 @@ def test_viewer_cannot_accept_suggestion(self):
+            'content', '<p>new content html</p>').to_dict()
+        self.resubmit_change_content = state_domain.SubtitledHtml(
+            'content', '<p>resubmit change content html</p>').to_dict()
+
+        suggestion_services.create_suggestion(
+            feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT, self.TARGET_TYPE,
+            self.exploration_id, self.target_version_id,
+            self.author_id,
+            self.change_dict, '')
+        suggestion = suggestion_services.query_suggestions(
+            [('author_id', self.author_id),
+             ('target_id', self.exploration_id)])[0]
+        self.suggestion_id = suggestion.suggestion_id
+        self.logout()
+
+    def test_author_can_update_suggestion(self):
+        self.login(self.author_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json('/mock/%s' % self.suggestion_id)
+        self.assertEqual(response['suggestion_id'], self.suggestion_id)
+        self.logout()
+
+    def test_owner_can_update_suggestion(self):
+        self.login(self.owner_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json('/mock/%s' % self.suggestion_id)
+        self.assertEqual(response['suggestion_id'], self.suggestion_id)
+        self.logout()
+
+    def test_reviewer_can_update_suggestion(self):
+        self.login(self.reviewer_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json('/mock/%s' % self.suggestion_id)
+        self.assertEqual(response['suggestion_id'], self.suggestion_id)
+        self.logout()
+
+    def test_non_author_cannot_update_suggestion(self):
+        self.login(self.user_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json(
+                '/mock/%s' % self.suggestion_id, expected_status_int=401)
+        self.assertEqual(
+            response['error'],
+            'You are not allowed to update the suggestion.')
+        self.logout()
+
+    def test_guest_cannot_update_suggestion(self):
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json(
+                '/mock/%s' % self.suggestion_id, expected_status_int=401)
+        self.assertEqual(
+            response['error'],
+            'You must be logged in to access this resource.')
+
+    def test_suggestions_with_invalid_suggestion_id_cannot_be_updated(self):
+        self.login(self.reviewer_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json(
+                '/mock/%s' % 'suggestion-id',
+                expected_status_int=400)
+        self.assertEqual(
+            response['error'], 'Invalid format for suggestion_id. ' +
+            'It must contain 3 parts separated by \'.\'')
+        self.logout()
+
+    def test_suggestions_with_no_suggestions_are_rejected(self):
```

Comment:
This name is slightly confusing. Perhaps instead: ``test_non_existent_suggestion_cannot_be_updated`` might be clearer?